dns_entry_checker/dns_entry_checker.py

```diff
@@ -0,0 +1,127 @@
+import paramiko
+from re import compile
+from getpass import getpass
+from collections import defaultdict
+
+
+# Creates an SSH client through Paramiko with the users details and returns the client
+def create_client(host, user, password):
+    client = paramiko.client.SSHClient()
+    client.set_missing_host_key_policy(paramiko.AutoAddPolicy)
+    client.connect(hostname=host, username=user, password=password, timeout=60)
+    return client
+
+
+# Executes a command through a Paramiko client, waits for an exit status and outputs the response
+def ssh_command(client, command):
+    _stdin, _stdout, _stderr = client.exec_command(command, get_pty=True, timeout=60)
+    _stdout.channel.recv_exit_status()
+    return _stdout.readlines()
+
+
+# Checks if the IP address and DNS match
+def pair_ip_and_dns(dns_pair, order_check_dict, ip_rexp):
+    last_rexp = compile(r"([0-9]{1,3}\.[0-9]{1,3}(?!.))")
+
+    ips = ip_rexp.findall(dns_pair)
+    ips = [ips[0][0], ips[1][0]]
+
+    order_values = last_rexp.findall(ips[1])[0].split(".")
+    order_check_dict[order_values[0]].append(order_values[1])
+    return ips
+
+
+# Checks if the DNS returns the matching IP
+def check_ip_dns_mismatch(
+    ips,
+    client,
+    ip_rexp,
+    backward_mismatch_file,
+    forward_mismatch_file,
+    backward_missing_file,
+):
+    if ips[0].replace("-", ".") == ips[1]:
+        returned_dns = ssh_command(client, "dig -x {s} +short".format(s=ips[1]))
+        if returned_dns:
+            backward_ip = ip_rexp.findall(returned_dns[0])
+            if not backward_ip:
+                backward_missing_file.write(f"{ips[1]}\n")
+            elif not backward_ip[0][0] == ips[0]:
+                backward_mismatch_file.write(f"{ips[1]}\n")
+        else:
+            backward_missing_file.write(f"{ips[1]}\n")
+    else:
+        forward_mismatch_file.write(f"{ips[1]}\n")
+
+
+# Checks which IPs are missing from 2-254
+def check_missing_ips(key, gap_missing_file, i):
+    res = set([eval(j) for j in key[1]])
+    missing_values = [k for k in sorted(set(range(2, 255))) if k not in res]
```

Comment:
why are you doing sorted(set(range(2, 255)))?
surely this works fine
```suggestion
    missing_values = [k for k in range(2, 255) if k not in res]
```